docs/changes/nddata/14438.feature.rst

```diff
@@ -0,0 +1,3 @@
+A more flexible and/or compact string representation is now available for
```

Comment:
```suggestion
A more flexible representation is now available for
```

I would drop the "and/or" and remove "string" because we have changed the `__repr__` but not `__str__`.